src/tvdb/client.py

```diff
@@ -97,56 +112,79 @@ def id(self, value: int | str) -> None:  # pyright: ignore[reportPropertyTypeMis
 
     @classmethod
     @abstractmethod
-    async def fetch(cls, media_id: int | str, *, client: "TvdbClient", extended: bool = False) -> "_Media": ...
-
+    def supports_meta(cls, meta: FetchMeta) -> bool:
+        """Check if the class supports a specific meta."""
+        ...
 
-@final
-class Movie(_Media):
-    """Class to interact with the TVDB API for movies."""
-
-    @override
     @classmethod
-    async def fetch(cls, media_id: int | str, client: "TvdbClient", *, extended: bool = False) -> "Movie":
+    async def fetch(
+        cls: type[Self],
```

Comment:
btw, this annotation is actually not even necessary when using `Self`, it should work with just:
```suggestion
        cls,
```

(Forgot about that when you asked about this on discord.)